src/api/variables-api.ts

```diff
@@ -4,15 +4,35 @@ import {
 } from "../interfaces/package-manager.interfaces";
 import {httpClientV2} from "../services/http-client-service.v2";
 import {FatalError} from "../util/logger";
+import {variableAssignmentApis} from "./variable-assignment-apis";
+import {URLSearchParams} from "url";
 
 class VariablesApi {
     public static readonly INSTANCE = new VariablesApi();
+    public static readonly ASSIGNMENT_APIS  = variableAssignmentApis;
+
 
     public async assignVariableValues(packageKey: string, variablesAssignments: VariablesAssignments[]): Promise<ContentNodeTransport[]> {
         return httpClientV2.post(`/package-manager/api/nodes/by-package-key/${packageKey}/variables/values`, variablesAssignments).catch(e => {
             throw new FatalError(`Problem updating variables of package ${packageKey}: ${e}`);
         });
     }
+
+    public async getCandidateAssignments(type: string, params: URLSearchParams): Promise<object[]> {
+        let apiUrl: string = VariablesApi.ASSIGNMENT_APIS[type]?.url;
```

Comment:
nit
```suggestion
        const apiUrl: string = VariablesApi.ASSIGNMENT_APIS[type]?.url;
```

Reply:
nvm I see it is being reassigned, though in that note I would suggest to try and keep it const ,so as to not change values of variables (arguably making it easier to separate code into pure functions in the future)

Reply:
I inverted the conditions and switched the variable to const.